app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +52,84 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // Prep vars
         TextView emptyText = findViewById(R.id.folderViewEmptyText);
+        TextView folderName = findViewById(R.id.folderNameText);
+        ImageButton newBut = findViewById(R.id.newPost);
+        ImageButton settingsBut = findViewById(R.id.folderSettings);
+        Button openBut = findViewById(R.id.buttonOpen);
+        Button editBut = findViewById(R.id.buttonEdit);
+        Button deleteBut = findViewById(R.id.buttonDelete);
         postRecyclerView = findViewById(R.id.posts_recycler_view);
-        currentUser = PosteApplication.getCurrentUser();
-        registerForContextMenu(postRecyclerView);
 
-        // Get current folder
-        int currentFolderId = getIntent().getIntExtra("folderId", -1);
-        try {
-            currentFolder = API.getFolderById(currentFolderId);
-
-            // Remove empty text if posts exist in folder
-            if (currentFolder.getPosts().size() > 0) {
-                emptyText.setVisibility(View.GONE);
+        // Create listeners for the folder buttons
+        newBut.setOnClickListener(view -> {
+            Post newPost = PosteApplication.getSelectedFolder().createNewPost(PosteApplication.getSelectedFolder());
```

Comment:
I think this is a good start. A few considerations:

- When a user clicks the button, this is going to send a request to create a new Post to the backend. However, this simply creates a post with the title 'Untitled', no URL, no description, etc. However...
- The API request looks good! You properly define an endpoint in MyApiService, and create a `CreatePost` request that sends a title, description, url, and folder. As a result...
- Clicking the 'new post' button should start an activity (a client-side page) for creating a post. This would retrieve the additional details: title, description, url. Then, when this is saved, the API call can be made with *all* the details (instead of creating an empty post with an 'Untitled' title)

Reply:
Agreed, Breaking up the API call is unnecessary, I Missunderstood and though you wanted me to split them up. 
In this case, since EditPost and create know poste are done from the same Activity Page then the API call should not be made here.
I will change it simple to open the EditPostActivity with no selected post. I will touch based with the student who refactored EditPostActivity to make sure we are passing the right data for the activity to know if it is pushing changes to the API or making a new Post in it.

Reply:
Exactly. We could even split up the activities -- even if they appear almost exactly identical -- as it may help abstracting out some concepts. For instance, while they appear the same to the user, creating a new post would have no fields filled out (blank slate) whereas editing a post should probably have the existing fields populated. Also, while creating a post is a straightforward, an edit would need a different API call to ensure we are truly editing the backend post and not creating a duplicate. It sounds like you know where to go with this, so I'll mark this as resolved.